packages/storage/src/storage.ts

```diff
@@ -833,6 +833,26 @@ export class Storage {
   }
 }
 
+/**
+ * Fetch the public read URL of a user file for the specified app.
+ * @param {String} path - the path to the file to read
+ * @param {String} username - The Blockstack ID of the user to look up
+ * @param {String} appOrigin - The app origin
+ * @param {String} zoneFileLookupURL - The URL
+ * to use for zonefile lookup. If falsey, this will use the
+ * blockstack.js's [[getNameInfo]] function instead.
+ * @return {Promise<string>} that resolves to the public read URL of the file
+ * or rejects with an error
+ */
+export function getUserAppFileUrl(
+  path: string,
+  username: string,
+  appOrigin: string,
+  zoneFileLookupURL?: string
+): Promise<string | undefined> {
```

Comment:
Would promote using an object as the argument when requiring this many parameters.

Reply:
Added these changes.